interfaces/builtin/network_setup_control.go

```diff
@@ -32,6 +32,19 @@ const networkSetupControlBaseDeclarationSlots = `
 const networkSetupControlConnectedPlugAppArmor = `
 # Description: Can read/write netplan configuration files
 
+# Allow use of the netplan binary from the base snap. With this interface, this 
+# is expected to be able to apply and generate new network configuration, as 
+# well as get information about the current network configuration.
+/usr/sbin/netplan ixr,
```

Comment:
it's a bit unclear what to do about that but it's not clear we can mandate for all bases to have neptlan inside them

Reply:
Well, I would expect this only to be used on Ubuntu Core systems, so really this is only a requirement of Ubuntu Core boot bases. We could make the snippet conditional and only applied on core systems if that makes you more comfortable?

Reply:
I understand the concern that the policy allows using the binary from the base snap, even if the base snap does not provide it, but I don't think this should be reason enough to block this pr because:
* there are no boot base snaps out there today that currently don't ship netplan
* the only base snaps that don't ship netplan are not boot base snaps (i.e. the freedesktop and the bare base snaps that I know of)
* if we consider what the effect of allowing a snap to be provided permission to run netplan is when there may eventually be a boot base snap that does not have netplan, the impact is very small, specifically if you built an app today (using this pr) on a base snap that has netplan, then you decide to switch base snaps, then your snap will no longer work, but I don't view this as a problem because we don't automatically move snaps from one base snap to another, and we don't allow base snaps to break the contract of providing the file.
* it's very unclear why someone would expect to be able to use network-setup-control on a non-boot base snap, i.e. what does it mean to be able to run netplan from the base snap when you have a freedesktop base snap? 

Really the concern here is about the interface contract providing access to something that might not be there, which is fine IMHO because there are plenty of other cases where this is true in interfaces such as when there is a resource in /dev/, we provide access via apparmor to the /dev/ file even if it doesn't exist. 

At the end of the day allowing this access will not break snaps and will not permit folks to build a snap that works then someday becomes broken. It only allows folks to knowingly build broken snaps, which they can already do today. Additionally this access addresses a product offering gap which is that in order to actually use netplan you have to essentially maintain your own netplan inside your snap which is a really unfortunate user story.

Reply:
let's say we consider netplan optional,  the other question is do we expect newer and older versions of netplan client to interoperate with newer/older version of actual system netplan?

I suppose we are ok because in the end it can only use the dbus interface that is the contract here? but actually we let it read the files as well under netplan? is the latter a problem?

Reply:
Well there are 2 cases:

1. with an older boot base and a newer app base snap
2. with a newer boot base and an older app base snap

For 1), the only way that the app snap could be broken is if the new netplan binary decides to send a different D-Bus API, which IMHO would be netplan breaking a compatibility guarantee, but I don't know if there exists such a version compatibility guarantee

For 2), the only way that the app snap could be broken is if the new netplan service implementation stops understanding old API, which is 100% a broken compatibility promise, and is the same kind of breakage that would happen without this PR if the client bundles and ships netplan in their app snap anyways (like they must be doing today anyways).

Thinking about this again, I think that if the netplan team says they do not guarantee forward compatibility such that a new netplan binary could some-day send incompatible data to an older netplan service, then we should not do this. I asked the team about this and will report back what I hear.

Regarding reading the files, I don't view this as a problem since really those are just python files underneath /usr/share/netplan/... that the netplan binary needs to read/load in order to work, as netplan doesn't install it's python files inside the typical site-packages directory which is allowed by the python apparmor abstraction AFAICT.